### PR TITLE
Always use forward slashes in SFTP (Fixes #334)

### LINKED ISSUE #334: Windows SFTP Path Handling
Trying to Backup to an SFTP repository on Windows like this:

    > .\restic.exe -r sftp://user@host//files/Backup/restic backup "C:\Users\Jacob1123"
    \files\Backup\restic\data does not exist

Note that the error message contains Backslashes instead of forward slashes.

Also, when I init a new repository via SFTP I get the following folders instead of a single folder "restic" containing the others:

    restic/
    restic\\keys/
    restic\\locks/
    restic\\data/
    restic\\snapshots/
    restic\\index/
    restic\\tmp/

I also tried using a minGW-Terminal, but got the same results.

### PR DESCRIPTION
**Pull request by JaCoB1123**
Add custom Join func that always uses forward slashes in SFTP

### CODE REVIEW

File: backend/sftp/sftp.go [hunk 1]
```diff
@@ -101,16 +102,16 @@ func Create(dir string, program string, args ...string) (*SFTP, error) {
...
+		Join(dir, backend.Paths.Snapshots),
+		Join(dir, backend.Paths.Index),
+		Join(dir, backend.Paths.Locks),
+		Join(dir, backend.Paths.Keys),
+		Join(dir, backend.Paths.Temp),
 	}
 
 	// test if config file already exists
-	_, err = sftp.c.Lstat(filepath.Join(dir, backend.Paths.Config))
+	_, err = sftp.c.Lstat(dir + backend.Paths.Config)
```
Comment (fd0) [inline backend/sftp/sftp.go]: Why did you remove the `path.Join()` here? is `dir` guaranteed to always end with a slash?

File: backend/sftp/sftp.go [hunk 2]
```diff
@@ -286,13 +287,33 @@ func (r *SFTP) Create() (backend.Blob, error) {
 	return &blob, nil
 }
 
+func Join(parts ...string) string {
```
Comment (fd0) [inline backend/sftp/sftp.go]: This function's body can be replaced by calling `path.Clean(path.Join(parts...))`.

File: backend/sftp/sftp.go [hunk 3]
```diff
@@ -286,13 +287,33 @@ func (r *SFTP) Create() (backend.Blob, error) {
...
+		result = result + "/" + s
+	}
+	return result
+}
+
 // Construct path for given backend.Type and name.
 func (r *SFTP) filename(t backend.Type, name string) string {
 	if t == backend.Config {
-		return filepath.Join(r.p, "config")
+		return Join(r.p + "config")
```
Comment (fd0) [inline backend/sftp/sftp.go]: Again, I think `+` may be wrong here, since `r.p` may not end with a slash. Call `Join(r.p, "config")` instead.

**Comment by fd0** [comment]
Thanks for your contribution! Please see my code review comments ;)

**Comment by JaCoB1123** [comment]
Thanks. I fixed the issues.
The missing joins were leftovers of my first try to fix it ;-)

**Comment by JaCoB1123** [comment]
Just noticed the Travis build failed. I can't figure out what the log says though.

**Comment by fd0** [comment]
This is a really strange issue that sometimes occurs on Darwin. I've restarted the test.

**Comment by fd0** [comment]
The test suceeded, thanks for your contribution!
